src/tactic/interactive.lean

```diff
@@ -635,5 +652,90 @@ do let ns := name_set.of_list xs,
      when (¬ ns.contains h.local_pp_name) $
        try $ tactic.clear h) ∘ list.reverse
 
+meta def format_names (ns : list name) : format :=
+format.join $ list.intersperse " " (ns.map to_fmt)
+
+private meta def format_binders : list name × binder_info × expr → tactic format
```

Comment:
It's too bad there isn't a function to check whether an expr is in the type class resolution cache. This approach won't notice if an instance has been added with `haveI` or `resetI`. But testing `is_class` will capture things like `nat.prime`, which is sometimes used as a type class and sometimes not.

Reply:
There's always room for manual tweaking